WalletWasabi/WabiSabi/WabiSabiCoordinator.cs

```diff
@@ -118,6 +118,31 @@ public void BanDescendant(object? sender, Block block)
 		}
 	}
 
+	public void BanDoubleSpenders(object? sender, Transaction tx)
+	{
+		// Detect and punish double spending coins
+		var disrupters = Arena.Rounds
```

Comment:
@lontivero 
This is dangerous. Arena and rounds are not designed to be touched from outside asynchronously. There is a read operation here and later you also close the round. 

The only place where we are touching rounds is Arena.ActionAsync and locking AsyncLock. 

You need to find a solution how to integrate this into Arena.